api/app/Models/PoolCandidate.php

```diff
@@ -411,4 +411,29 @@ public function isDraft()
     {
         return is_null($this->submitted_at) || $this->submitted_at->isFuture();
     }
+
+    /**
+     * Scope the query to PoolCandidate's the current user can view
+     */
+    public function scopeAvailableToQuery(Builder $query)
+    {
+        $userId = Auth::user()->id;
+        $user = User::find($userId);
+
+        if ($user->isAbleTo("view-any-submittedApplication")) {
+            return $query; //We filter out drafts in another scope
```

Comment:
If we implement a scope like this which scope to items the user is authorized to view, I think it should cover all cases, including the logic around drafts.

Reply:
I think this makes sense since the `notDraft` scope was introduced for an incredibly similar use case 👍 

[update scope to filter draft, restrict viewing own](https://github.com/GCTC-NTGC/gc-digital-talent/pull/6064/commits/4b95d1f03c3c21c8d35155b715741e7fcb796fbd)